addon-test-support/-private/mock-window.js

```diff
@@ -4,8 +4,16 @@ export function mockWindow(scope) {
   if (!scope) {
     throw new Error('mockWindow must be called with `this` as the first function parameter!');
   }
-  if (!scope.register) {
-    throw new Error('mockWindow must be called from an integration test!');
+  
+  let register = scope.register;
+  
+  if(scope.owner && scope.owner.register) {
+    register = scope.owner.register;
   }
-  return scope.register('service:window', windowMockFactory(), { instantiate: false });
+  
+  if (!register) {
+    throw new Error('mockWindow must be called from an unit/integration test!');
+  }
+  
+  return register('service:window', windowMockFactory(), { instantiate: false });
```

Comment:
Tests are failing. Maybe because `register` is now called as simple function, and not as a method, so lacks the correct `this` context. Maybe use `register.call(this, 'service:window', ...)` here?

Reply:
Should it be `this` or `scope`?